### DISCUSSION TITLE
JSON date properties are not hydrated back as Dates

### DISCUSSION THREAD

**Discussion by alexandre-abrioux** [mikro-orm/mikro-orm#6077] [General]
### Describe the bug

This regression was introduced in v6.3.11 and probably linked to https://github.com/mikro-orm/mikro-orm/commit/55df57ff1aa84e5d45188d849ba09e91ae6d3642 (https://github.com/mikro-orm/mikro-orm/issues/6050).

When JSON properties contain dates, they were previously hydrated back as Javascript Dates. Now, they are hydrated as strings.

### Reproduction

Here is a test case: https://github.com/alexandre-abrioux/mikro-orm-issues/blob/iss-6076/src/example.test.ts

### What driver are you using?

@mikro-orm/sqlite

### MikroORM version

6.3.11

### Node.js version

22.4.0

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
I'll check what happened, but your test case should behave this way, there should be no Date hydration on JSON properties, they are not processed anyhow, we just parse the JSON. If this was not the case before, I would even call the previous way a bug.

You need embeddables if you want to control how the properties are hydrated.

**Comment by B4nan** [comment]
Your test case behaves the same in v6.3.10, as well as on v6.0.0.

**Comment by alexandre-abrioux** [comment]
Sorry @B4nan, I should have run the test with v6.3.10 before opening the issue. Something broke for us after updating from 6.3.10 to 6.3.11. I will dig deeper and open another issue if I find something relevant. Sorry for taking up your time and thanks again for your help 🙏

**Reply by B4nan** [reply]
Keep in mind that this is not how things are supposed to be working. If you saw a `Date` object inside a JSON property somewhere, it was set by you, not hydrated from the database by the ORM.

**Reply by alexandre-abrioux** [reply]
I found out why we had dates in the first place 🙂 We use the MongoDB driver, but in the test case, I was using the SQLite driver. The MongoDB driver indeed returns ISODates objects (BSON dates) when retrieving documents. I've opened another issue here: https://github.com/mikro-orm/mikro-orm/issues/6078
Thanks again for your help!
